### PR TITLE
Add "message wall"

### PR DESCRIPTION
**Pull request by pbartela2**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules
+public
+npm-debug.log*
+.vscode
+.history
```
Comment (kjendrzyca) [inline .gitignore]: better to use https://www.gitignore.io/api/node

File: package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "react-nauka",
```
Comment (kjendrzyca) [inline package.json]: english plz :)

File: .eslintrc.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+extends: 
+  - "standard-react"
+env:
+  jest: true
```
Comment (kjendrzyca) [inline .eslintrc.yml]: not using jest

File: package.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  "devDependencies": {
+    "css-loader": "^0.28.4",
+    "eslint": "^4.0.0",
+    "eslint-config-standard": "^10.2.1",
+    "eslint-config-standard-react": "^5.0.0",
+    "eslint-plugin-promise": "^3.5.0",
+    "eslint-plugin-react": "^7.1.0",
+    "eslint-plugin-standard": "^3.0.1",
+    "file-loader": "0.11.2",
+    "jest": "^20.0.4",
```
Comment (kjendrzyca) [inline package.json]: not using jest

**Comment by kjendrzyca** [comment]
would be better to simplify - remove less and stick to css

File: package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    "less-loader": "^4.0.4",
+    "style-loader": "^0.18.2",
+    "url-loader": "0.5.9"
+  },
+  "dependencies": {
+    "babel-core": "^6.25.0",
+    "babel-loader": "^7.1.0",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
+    "classnames": "2.2.5",
```
Comment (kjendrzyca) [inline package.json]: is it used?

File: src/app.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+class App extends Component {
+  render () {
+    console.log(this.props);
```
Comment (kjendrzyca) [inline src/app.js]: remove console.log

File: .gitignore (hunk 1 above)
Comment (pbartela2) [inline .gitignore]: Done

File: src/components/TextArea/TextAreaComponent.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import './TextArea.less';
+
+class TextArea extends Component {
+
```
Comment (kjendrzyca) [inline src/components/TextArea/TextAreaComponent.js]: remove empty line

File: src/helpers/imageEmbeder.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        __html: rawText.replace(urlRegex, function (url) {
+            
+            if ((url.indexOf(".jpg") > 0) || (url.indexOf(".png") > 0) || (url.indexOf(".gif") > 0)) {
+                return '<img src="' + url + '">' + '<br/>'
+            } else {
+                return '<a href="' + url + '">' + url + '</a>' + '<br/>'
+            }
+        })
+    }
+}
```
Comment (kjendrzyca) [inline src/helpers/imageEmbeder.js]: could be rewritten to plain react, will try to do this today

File: src/index.html [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+<html>
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+  <title>Training</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <style>
+    body {
+      margin: 0;
+        padding: 0;
```
Comment (kjendrzyca) [inline src/index.html]: indent

File: src/scenes/Main/index.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    }
+
+    componentDidMount() {
+        const posts = loadArray(storageKey);
+        posts && this.setState({posts: posts});
+    }
+
+    handleSubmit(post) {
+        const { posts } = this.state;
+        posts.push(post);
```
Comment (kjendrzyca) [inline src/scenes/Main/index.js]: use concat, be brave

File: webpack.config.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+                limit: 25000
+            }
+        }]
+    },
+
+    resolve: {
+        extensions: ['.js']
+    },
+
+    plugins: [
```
Comment (kjendrzyca) [inline webpack.config.js]: dump this plugin, for simplicity, will do this next time

File: package.json (hunk 2 above)
Comment (pbartela2) [inline package.json]: Sad copy paste... ;<

File: package.json (hunk 4 above)
Comment (pbartela2) [inline package.json]: Done

File: .eslintrc.yml (hunk 3 above)
Comment (pbartela2) [inline .eslintrc.yml]: done

File: src/app.js (hunk 6 above)
Comment (pbartela2) [inline src/app.js]: Done

File: src/index.html (hunk 9 above)
Comment (pbartela2) [inline src/index.html]: Moved to css

File: src/scenes/Main/index.js (hunk 10 above)
Comment (pbartela2) [inline src/scenes/Main/index.js]: Ok ;(

File: webpack.config.js (hunk 11 above)
Comment (pbartela2) [inline webpack.config.js]: Done

File: package.json (hunk 5 above)
Comment (pbartela2) [inline package.json]: Yes. In "Post" component

File: src/helpers/imageEmbeder.js (hunk 8 above)
Comment (pbartela2) [inline src/helpers/imageEmbeder.js]: ok

File: src/components/TextArea/TextAreaComponent.js (hunk 7 above)
Comment (pbartela2) [inline src/components/TextArea/TextAreaComponent.js]: Done

File: src/scenes/Main/index.js (hunk 10 above)
Comment (kjendrzyca) [inline src/scenes/Main/index.js]: and add `this.setState({posts})`

File: src/components/Post/PostComponent.js [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    constructor(props) {
+        super(props);
+        this.handleVisibilityClick = this.handleVisibilityClick.bind(this);
+        this.state =  {
+            visible: true
+        }
+    }
+
+    handleVisibilityClick() {
+        this.setState({visible: !this.state.visible});
```
Comment (kjendrzyca) [inline src/components/Post/PostComponent.js]: If you rely on some previous state use this way of changing state:
```
this.setState(({visible}) => ({visible: !visible})
```

File: src/components/Posts/PostsComponent.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Post } from './../Post/';
+const Posts = props => {
+    const { posts } = props;
+    const postRender = [];
+    posts.forEach(function (post, index) {
```
Comment (kjendrzyca) [inline src/components/Posts/PostsComponent.js]: use `map`, `for` and `forEach` is not the react way

**Pull request merged**